cachi2/core/scm.py

```diff
@@ -62,7 +62,21 @@ def get_repo_id(repo: Union[str, PathLike[str], Repo]) -> RepoID:
 
 def _canonicalize_origin_url(url: str) -> str:
     if "://" in url:
-        return url
+        parsed_url: ParseResult = urlparse(url)
+        cleaned_netloc = parsed_url.netloc.replace(
+            f"{parsed_url.username}:{parsed_url.password}@",
+            "",
+        )
+        return urlunparse(
+            (
+                parsed_url.scheme,
+                cleaned_netloc,
+                parsed_url.path,
+                parsed_url.params,
+                parsed_url.query,
+                parsed_url.fragment,
+            )
+        )
```

Comment:
Can we simplify this to something like the following?

```
return urlparse(url)._replace(netloc="").geturl()
```
(you can even split this to multiple lines if you think ^this lacks readability)
`urlparse` returns a [named tuple](https://docs.python.org/3.12/library/urllib.parse.html#urllib.parse.urlparse) and named tuples can be modified via the [`_replace`](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace) method. Seems much cleaner to me.

Reply:
Actually, there are multiple parts in netloc, so more like this
```
parsed = urlparse(url)
return parsed._replace(netloc=parsed.netloc.split("@")[1]).geturl()
```

Reply:
A few were failing with this approach...
How about current solution ?
```
>           assert repo_id.origin_url == expect_result
E           AssertionError: assert 'ssh://git.host.com/some/path' == 'ssh://user@git.host.com/some/path'
E
E             - ssh://user@git.host.com/some/path
E             ?       -----
E             + ssh://git.host.com/some/path

tests/unit/test_scm.py:51: AssertionError
```

Reply:
Sorry, missed that detail, your adjustment certainly works.